src/inc/torelease.h

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This class acts a smart pointer which calls the Release method on any object
+// you place in it when the ToRelease class falls out of scope.  You may use it
+// just like you would a standard pointer to a COM object (including if (foo),
+// if (!foo), if (foo == 0), etc) except for two caveats:
+//     1. This class never calls AddRef and it always calls Release when it
+//        goes out of scope.
+//     2. You should never use & to try to get a pointer to a pointer unless
+//        you call Release first, or you will leak whatever this object contains
+//        prior to updating its internal pointer.
+template<class T>
+class ToRelease
```

Comment:
I don't like the name either. I moved this template from SOS; in SOS it has to be a different name not to conflict with the runtime one in inc\holder.h.  SOS and createdump can't use the runtime holder because it pulls in and depends on other runtime macros and templates.

I renamed the file/macro to releaseholder.h/ReleaseHolder<> but backed out my change to SOS's util.h to include its ToRelease<> macro.